core/src/test/java/com/graphhopper/routing/util/BikeFlagEncoderTest.java

```diff
@@ -68,23 +68,40 @@ public void testGetSpeed()
         way.clearTags();
         way.setTag("highway", "footway");
         way.setTag("bicycle", "yes");
-        assertEquals(6, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
         way.clearTags();
         way.setTag("highway", "footway");
         way.setTag("surface", "paved");
         way.setTag("bicycle", "yes");
 
-        assertEquals(6, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
         way.clearTags();
+        way.setTag("highway", "cycleway");
+        assertEquals(18, encoder.getSpeed(way));
+        assertPriority(VERY_NICE.getValue(), way);
+        int cyclewayspeed = encoder.getSpeed(way);
+
+        // Make sure that highway=cycleway" and highway=path with bicycle = designated give same result
+        way.clearTags();
         way.setTag("highway", "path");
+        way.setTag("bicycle", "designated");
+        assertEquals(cyclewayspeed, encoder.getSpeed(way));
+        assertPriority(VERY_NICE.getValue(), way);
+
         way.setTag("bicycle", "yes");
-        assertEquals(12, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
+        way.setTag("surface", "unpaved");
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
+
+        way.setTag("surface", "paved");
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
+
         // Pushing section Ok !!
```

Comment:
Sorry, I forgot. I believe `// Pushing section Ok !!`meant that the next test
```
        way.setTag("highway", "path");
        way.setTag("surface", "paved");
```
 is fine in case that the result is that it is a pushing section. Please delete the comment